src/Chapter3.hs

```diff
@@ -970,6 +1092,28 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
+data DaysOfWeek = Monday 
+                  | Thuesday 
+                  | Wednesday 
+                  | Thursday 
+                  | Friday 
+                  | Saturday 
+                  | Sunday deriving (Show, Eq, Enum)
+
+isWeekend :: DaysOfWeek -> Bool
+isWeekend Saturday = True
+isWeekend Sunday = True
+isWeekend _ = False
+
+nextDay :: DaysOfWeek -> DaysOfWeek
+nextDay d
+    | d == Sunday = Monday
```

Comment:
That is right!
You could also use the instance of `Bounded` in here:

```suggestion
    | d == maxBound = minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂